junebug/stores.py

```diff
@@ -142,3 +151,46 @@ def get_statuses(self, channel_id):
             (k, TransportStatus.from_json(v))
             for k, v in statuses.iteritems()
         ))
+
+
+class MessageRateStore(BaseStore):
+    '''Gets called everytime a message should be counted, and can return the
+    current messages per second.'''
+
+    def get_seconds(self):
+        return time.time()
+
+    def get_key(self, channel_id, label, bucket):
+        return super(MessageRateStore, self).get_key(
+            channel_id, label, str(bucket))
+
+    def _get_current_key(self, channel_id, label, bucket_size):
+        bucket = int(self.get_seconds() / bucket_size)
+        return self.get_key(channel_id, label, bucket)
+
+    def _get_last_key(self, channel_id, label, bucket_size):
+        bucket = int(self.get_seconds() / bucket_size) - 1
+        return self.get_key(channel_id, label, bucket)
+
+    def increment(self, channel_id, label, bucket_size):
+        '''Increments the correct counter. Should be called whenever a message
+        that should be counted is received.
+
+        Note: bucket_size should be kept constant for each channel_id and label
+        combination. Changing bucket sizes results in undefined behaviour.'''
+        key = self._get_current_key(channel_id, label, bucket_size)
+        self.ttl = bucket_size * 2
```

Comment:
I originally wanted to inherit as much as possible from the `BaseStore`, but I see now that it is hurting us more than helping us.

The reason I wanted `bucket_size` to be passed per function is that in the API when we're fetching the msgs/s, `inbound` and `outbound` could be configured to have different bucket sizes. This reason is why we had `channel_id` being passed to every function for the other stores, instead of setting the channel id in the constructor.

The reason that `ttl` is being set here, is because we want to set the ttl when we increment, but we don't want to set the ttl again when we fetch the value (there we set `ttl` to `None`, so that we don't call `EXPIRE`.